src/Zynga/Framework/PgData/V1/PgRow/PkComesFromMemcache.hh

```diff
@@ -109,11 +111,13 @@ abstract class PkComesFromMemcache extends PgRow {
 
   public function createPkKeyForMC(): string {
     $writeDatabase = $this->pgModel()->db()->getWriteDatabase();
-
-    $connectionString = '';
+    $shardId = 0;
+    if($writeDatabase instanceof ShardedDriverInterface) {
+      $shardId = $writeDatabase->getConfig()->getShardId($writeDatabase->getShardType());
+    }
 
     $tableName = $this->getTableName();
-    $pkKey = 'pgd:'.hash("sha256", $connectionString).'|'.$tableName.':pk';
+    $pkKey = 'pgd:'.$shardId.'|'.$tableName.':pk';
```

Comment:
iirc, we kind of discussed this topic in the past to avoid mc key overlaps across diff schemas with same table names & even tried remove md5 hash. we can leave connection string hash too in this combo if it's not too much as a safe check.